src/main/java/group1/mips_simulator/components/CPU.java

```diff
@@ -0,0 +1,8 @@
+package group1.mips_simulator.components;
+
+public class CPU {
+    RegisterFile Registers = new RegisterFile();
```

Comment:
Will the CPU be a component of the MipsComputer class? Like will the computer "have a" CPU? 

Also, variable names are usually starting with a lower case letter. So consider the following change:
```suggestion
    RegisterFile registers = new RegisterFile();
```

Reply:
woops, will fix the capitalizations

So, the way im thinking of it, and i have a suggestion for you on your end too (i can make it a more formal request after this, ill find the actual branch where you made this):

- could you rename the "MipsComputer" class to just "Computer"?

- a Computer utilizes a set of Hardware components (things like CPU, Memory, Busses, etc.) to operate on data
- each of these hardware components have their own sets of hardware, subcomponents, and operations they need to perform (i.e. the CPU contains registers, basic instruction set, a ROM loader; CPU needs to perform elements to execute the Instr set, read ROM contents to memeory, etc.) These are the attributes and responsibilities of the CPU
- I envision the Computer is almost just a collection of these components, and maybe the interface to the Operator+FieldEngi Consoles (or UI); 
i.e. we use the Computer class to configure our computer ("I want 1 cpu, 1 memory, 1 L2 cache) and be its interface to the UI, we'd define User operations and route them through here

this is just how im thinking of it, im very open to alternative architecture

Reply:
Sounds good to me! Just wanna make sure I'm keeping up with the architecture as it evolves :)